src/frontend/src/optimizer/plan_visitor/max_one_row_visitor.rs

```diff
@@ -92,3 +96,42 @@ impl PlanVisitor<bool> for HasMaxOneRowApply {
         plan.max_one_row() | self.visit(plan.left()) | self.visit(plan.right())
     }
 }
+
+pub struct CountRows;
+
+impl PlanVisitor<Option<usize>> for CountRows {
```

Comment:
Can we also add count for Agg? From what I know, agg has a possibility of produce 0 rows, so no.

Reply:
Agg without group key always produces exact one row.

Reply:
Hmm, ok, in that case we can count it as well, let me add test with 0 values below the agg

Reply:
Oops, we have a bug:
```sql
dev=> create table(v1 int);
CREATE_TABLE
dev=> create materialized view v2 as select sum(v1) from t;
CREATE_MATERIALIZED_VIEW
dev=> select * from v2;
 sum 
-----
(0 rows)

dev=> 
```

Reply:
Oops :dizzy_face: 
```sql
dev=> create materialized view v3 as select count(*) from t;
CREATE_MATERIALIZED_VIEW
dev=> select * from v3;
 count 
-------
(0 rows)
```

Reply:
Oops
```sql
dev=> create materialized view v4 as select sum(v1) from t group by v1;
CREATE_MATERIALIZED_VIEW
dev=> select * from v4;
 sum 
-----
(0 rows)
```

Reply:
I will fix it.

Reply:
For `SimpleAgg`, We have a known bug which is that it misses the first initial row.

Reply:
Fixed: https://github.com/risingwavelabs/risingwave/pull/7932

Let's merge that and then count the exact rows for simple agg here.